apps/vtex-my-subscriptions-3/react/components/ProductImage/index.tsx

```diff
@@ -14,46 +14,41 @@ class ProductImage extends PureComponent<Props> {
   }
 
   public state = {
-    isLoading: true,
-    hasLoadError: false,
+    displayPlaceholder: true,
   }
 
   private handleStopLoading = () => {
-    setTimeout(() => this.setState({ isLoading: false }), 100)
+    this.setState({ displayPlaceholder: false })
   }
 
   private handleError = () => {
-    this.handleStopLoading()
-    this.setState({ hasLoadError: true })
+    this.setState({ displayPlaceholder: true })
```

Comment:
In case of error, will the placeholder continue to appear? Doesn't that give the feeling that the page is still loading?
In the past implementation apparently in case of error the image was not displayed and neither was the placeholder.

Reply:
You're right about the current behavior, we are showing the placeholder in error cases. But I believe this was also old behavior. `displayPaceholder` was a derived state, and was set to `true` due to `hasLoadError` also being `true`

```js
const displayPaceholder = isLoading || hasLoadError || !hasImage
```

Reply:
Ah, ok! Make sense to me.